File: Matrixes/tridiagonal.js
module.exports = (a, b, c, d) => {
    N = d.length
    c[0] = c[0] / b[0];
    d[0] = d[0] / b[0];

    for (i = 1; i < N; i++) {
        temp = b[i] - a[i] * c[i - 1];
        if(i < (N-1)) c[i] = c[i] / temp;
        d[i] = (d[i] - a[i] * d[i - 1]) / temp;
        console.log('d: '+d)
    }

    x = d
    for (i = N - 2; i >= 0; i--){
        x[i] = d[i] - c[i] * x[i + 1];
        console.log('d: '+x)
    }
        
    return x;
}

File: Matlab/tridiagonal.m
a = input('Enter A')
b = input('Enter B')
c = input('Enter C')
d = input('Enter D')
%a, b, c son los vectores columnas para la matriz tridiagonal comprimida, d es el vector de la derecha
% N is el numero de filas
N = length(d);
 
% modifica los coeficientes de la primera fila
c(1) = c(1) / b(1); % riesgo de division por cero
d(1) = d(1) / b(1); 
 
for n = 2:1:N
    temp = b(n) - a(n) * c(n - 1);
    if (n<N)
        c(n) = c(n) / temp;
    end
    d(n) = (d(n) - a(n) * d(n - 1)) / temp;
end
 
% ahora sustituimos
x(N) = d(N);
for n = (N - 1):-1:1
    x(n) = d(n) - c(n) * x(n + 1);
end